### PR TITLE
Fixed Makefiles manually to get around linking errors when using an LLVM with terminfo.

### PR DESCRIPTION
**Pull request by SanderVocke**
Regarding Issues:
#1112 
#2156 

This is not a complete fix: only the app Makefiles have been patched. Cmake should be done as well. I don't have a working CMake build to do and test that on.

### CODE REVIEW

**Comment by abadams** [comment]
LLVM_LDFLAGS adds -Lllvm-build/lib, which could let us accidentally have
compiled generators depend on linking LLVM, which we really don't want.

I'd instead do something like:

LDFLAGS += $(shell $(LLVM_CONFIG) --system_libs)

in apps/support/Makefile.inc

**Comment by SanderVocke** [comment]
How about something like this? I didn't add it to LDFLAGS in the support makefile, since then it would also be included in rules that don't need these libs. But I made a new LLVM_SYSTEM_LDFLAGS in the support makefile, and added it to app Makefiles where necessary.

**Comment by steven-johnson** [comment]
Travis build looked like a flake; I restarted it manually.

**Comment by steven-johnson** [comment]
Conceptually this fix looks good; to be super-nitpicky, it might be good to rename LLVM_SYSTEM_LDFLAGS to something like HALIDE__SYSTEM_LDFLAGS; even though these dependencies come by way of LLVM, the client at this point shouldn't have to deal with anything named LLVM.

**Comment by SanderVocke** [comment]
That makes sense - fixed. 
I wonder if _SYSTEM_ should still be in the name - that kind of came from the fact that these flags came from calling `llvm-config --system-libs`. But I suppose these can be called "Halide's system libs' from this perspective.

**Comment by steven-johnson** [comment]
LGTM. @abadams, any concerns?

File: apps/c_backend/Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ $(BIN)/pipeline_c.cpp: $(BIN)/pipeline_exec
 $(BIN)/run: run.cpp $(BIN)/pipeline_c.cpp $(BIN)/pipeline_native.a
 	$(CXX) $(CXXFLAGS) -Wall -I$(BIN) $(filter-out %.h,$^) -o $@  $(LDFLAGS)
 
-$(BIN)/pipeline_cpp_exec: pipeline_cpp_generator.cpp $(GENERATOR_DEPS)
+$(BIN)/pipeline_cpp_exec: pipeline_cpp_generator.cpp $(GENERATOR_DEPS) $(HALIDE_SYSTEM_LDFLAGS)
```
Comment (abadams) [inline apps/c_backend/Makefile]: Was this added on the right line?

Comment (SanderVocke) [inline apps/c_backend/Makefile]: Oops, misplaced that one. Fixed below.

**Pull request merged**